### PR TITLE
Enable serialization of user attributes of a DockContent

### PR DESCRIPTION
**Pull request by ahuarte47**
This change enables the serialization of user attributes of a IDockContent interface.

### CODE REVIEW

File: WinFormsUI/Docking/DockPanel.Persistor.cs [hunk 1]
```diff
@@ -93,6 +93,13 @@ public bool IsFloat
                     get { return m_isFloat; }
                     set { m_isFloat = value; }
                 }
+
```
Comment (lextm) [inline WinFormsUI/Docking/DockPanel.Persistor.cs]: Is it possible to avoid changing this file?

**Comment by lextm** [comment]
I think all changes are OK as it demonstrates how to load/save custom things. But I don't think `DockPanel.Persistor.cs` should be changed.

File: WinFormsUI/Docking/DockPanel.Persistor.cs (hunk 1 above)
Comment (ahuarte47) [inline WinFormsUI/Docking/DockPanel.Persistor.cs]: Hi, I attached the IDockContent to its ContentStruct object (private struct in Dockpanel.Persistor.cs) when it is created to enable send the XmlReader object to the DockContent managed.

Would you like other option ?

Best Regargs
Alvaro

**Comment by roken** [comment]
Custom serialization is purposefully handled outside of the DockPanel persistence; this was not by accident:  https://github.com/dockpanelsuite/dockpanelsuite/wiki/LayoutPersistence

Embedding user data in the XML for the DockPanel persistence makes serialization of the DockPanel extremely fragile.  I would vote we decline this pull request and custom persistence should be handled as a concern of the application using the hooks provided in the wiki link above.

**Comment by ahuarte47** [comment]
Hy Ryan, I agree that it is possible user serialization using the DeserializeDockContent delegate. IMHO this new optional persistance allow work with XmlReader/XmlWriter for persist more complex structured data in a more comfortable style that parse manually a single string.

Best Regads
Alvaro

**Comment by ahuarte47** [comment]
Hi, I added serialization of global user attributes of a DockPancel

**Comment by ahuarte47** [comment]
Hi, I close this very old pull. Thanks for your reviews!

**Pull request closed without merging**